app/models/products/user_note_mode.rb

```diff
@@ -0,0 +1,49 @@
+module Products
+
+  class UserNoteMode
+
+    attr_reader :raw_value
+
+    def self.values
+      valid_values.map { |value| new(value) }
+    end
+
+    def self.valid_values
+      %w(hidden optional required).freeze
+    end
+
+    def initialize(raw_value)
+      @raw_value = raw_value
+      freeze
```

Comment:
It will initialize the object first. Then when you call `valid?` or `save` on the model, that will check if the inclusion list `include?` this object. I have an idea that I may play with later.